doc/source/ray-air/examples/torch_image_example.md

```diff
@@ -0,0 +1,255 @@
+
+train_dataset
+```
+
+## Train a convolutional neural network
+
+```{code-cell} python3
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class Net(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = nn.Conv2d(3, 6, 5)
+        self.pool = nn.MaxPool2d(2, 2)
+        self.conv2 = nn.Conv2d(6, 16, 5)
+        self.fc1 = nn.Linear(16 * 5 * 5, 120)
+        self.fc2 = nn.Linear(120, 84)
+        self.fc3 = nn.Linear(84, 10)
+
+    def forward(self, x):
+        x = self.pool(F.relu(self.conv1(x)))
+        x = self.pool(F.relu(self.conv2(x)))
+        x = torch.flatten(x, 1)  # flatten all dimensions except batch
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+```
+
+We define our training logic in a function called `train_loop_per_worker`.
+
+`train_loop_per_worker` contains regular PyTorch code with a few notable exceptions:
+* We wrap our model with `train.torch.prepare_model`.
+* We call `get_dataset_shard` and `to_torch` to convert a subset of our training data to a Torch dataset.
+* We save model state using `train.save_checkpoint`.
+
+```{code-cell} python3
+from ray import train
+import torch.optim as optim
+
+
+def train_loop_per_worker(config):
+    model = train.torch.prepare_model(Net())
+    
+    criterion = nn.CrossEntropyLoss()
+    optimizer = optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
+
+    train_dataset_shard: torch.utils.data.Dataset = train.get_dataset_shard("train").to_torch(
+        feature_columns=["image"],
+        label_column="label",
+        batch_size=config["batch_size"],
+        unsqueeze_feature_tensors=False,
+        unsqueeze_label_tensor=False
+    )
+
+    for epoch in range(2):
+        running_loss = 0.0
+        for i, data in enumerate(train_dataset_shard):
+            # get the inputs; data is a list of [inputs, labels]
+            inputs, labels = data
+
+            # zero the parameter gradients
+            optimizer.zero_grad()
+
+            # forward + backward + optimize
+            outputs = model(inputs)
+            loss = criterion(outputs, labels)
+            loss.backward()
+            optimizer.step()
+
+            # print statistics
+            running_loss += loss.item()
+            if i % 2000 == 1999:  # print every 2000 mini-batches
+                print(f"[{epoch + 1}, {i + 1:5d}] loss: {running_loss / 2000:.3f}")
+                running_loss = 0.0
+
+        train.save_checkpoint(model=model.module.state_dict())
+```
+
+Finally, we can train our model. This should take a few minutes to run.
+
+```{code-cell} python3
+from ray.ml.train.integrations.torch import TorchTrainer
+
+trainer = TorchTrainer(
+    train_loop_per_worker=train_loop_per_worker,
+    train_loop_config={"batch_size": 2},
+    datasets={"train": train_dataset},
+    scaling_config={"num_workers": 2}
+)
+result = trainer.fit()
+latest_checkpoint = result.checkpoint
+```
```

Comment:
I've added a small paragraph. Lemme know what you think.

Anecdotally, it's difficult to create a Ray Cluster and scale out, so I'm not sure if we should attempt to explain that in this "Hello World" example. It might be better to just leave a sentence like:

> To scale your training script, read ...

I feel like to appropriately explain scaling out, we'd need to add a lot of information that's out of scope for this example.